warehouse/integrations/datalake/schema-repository/schema_repository.go

```diff
@@ -42,11 +43,31 @@ type SchemaRepository interface {
 	CreateTable(tableName string, columnMap map[string]string) (err error)
 	AddColumns(tableName string, columnsInfo []warehouseutils.ColumnInfo) (err error)
 	AlterColumn(tableName, columnName, columnType string) (err error)
+	RefreshPartitions(tableName string, loadFiles []warehouseutils.LoadFileT) error
+}
+
+func UseGlue(w *warehouseutils.Warehouse) bool {
+	glueConfig := warehouseutils.GetConfigValueBoolString(UseGlueConfig, *w)
+	hasAWSRegion := misc.HasAWSRegionInConfig(w.Destination.Config)
+	return glueConfig == "true" && hasAWSRegion
 }
 
 func NewSchemaRepository(wh warehouseutils.Warehouse, uploader warehouseutils.UploaderI) (SchemaRepository, error) {
-	if warehouseutils.GetConfigValueBoolString(UseGlueConfig, wh) == "true" && misc.HasAWSRegionInConfig(wh.Destination.Config) {
+	if UseGlue(&wh) {
 		return NewGlueSchemaRepository(wh)
 	}
 	return NewLocalSchemaRepository(wh, uploader)
 }
+
+// LoadFileBatching batches load files for refresh partitions
+func LoadFileBatching(files []warehouseutils.LoadFileT, batchSize int) [][]warehouseutils.LoadFileT {
+	fileBatches := make([][]warehouseutils.LoadFileT, 0, len(files)/batchSize+1)
+
+	for len(files) > 0 {
+		cut := int(math.Min(float64(len(files)), float64(batchSize)))
```

Comment:
In general would avoid using float64 for anything that requires precision: https://jvns.ca/blog/2023/01/13/examples-of-floating-point-problems/#example-1-the-odometer-that-stopped

However, I can not think of an example where this would fail. So take this as a minor suggestion:
```suggestion
		cut := batchSize
		if len(files) > cut {
		    cut = len(files)
		}
```

Reply:
Thanks for pointing that out. @lvrach

Reply:
For some reason, I was expecting the math package to have `func max(x, y int) int`. Although there was one more [package](https://github.com/pkg/math) by Dave Cheney it's archived not. There is [proposal](https://github.com/golang/go/issues/50019) to have a generic implementation for this.